File: src/main/java/br/com/tech/challenge/service/validation/vote/AssociateAlreadyVoted.java
package br.com.tech.challenge.service.validation.vote;


import br.com.tech.challenge.domain.VoteDetail;
import br.com.tech.challenge.exceptions.AssociateAlreadyVotedException;
import br.com.tech.challenge.repository.VoteRepository;
import lombok.AllArgsConstructor;
import org.springframework.core.annotation.Order;
import org.springframework.stereotype.Service;

@Service
@Order(2)
@AllArgsConstructor
public class AssociateAlreadyVoted implements VoteValidation {
    private VoteRepository voteRepository;
    @Override
    public void validate(VoteDetail voteDetail) {
        var result = voteRepository.existsVoteFromAssociateAndSession(
                voteDetail.session().getId(), voteDetail.associateDetail().cpf());

        if(result){
            throw new AssociateAlreadyVotedException(STR."Associate with {CPF} = \{voteDetail.associateDetail().cpf()} already vote");
        }
    }
}

File: src/main/java/br/com/tech/challenge/service/validation/vote/AssociateAlreadyVoted.kt
package br.com.ntconsult.challenge.service.validation.vote

import br.com.ntconsult.challenge.domain.VoteDetail
import br.com.ntconsult.challenge.exception.AssociateAlreadyVotedException
import br.com.ntconsult.challenge.repository.VoteRepository
import org.springframework.core.annotation.Order
import org.springframework.stereotype.Service

@Service
@Order(2)
class AssociateAlreadyVoted(private val voteRepository: VoteRepository): VoteValidation {
    override fun valid(voteDetail: VoteDetail) {
        val result = voteRepository.existsVoteFromAssociateAndSession(
                voteDetail.session.id!!, voteDetail.associateDetail.cpf)

        if(result){
            throw AssociateAlreadyVotedException("Associate with {CPF} = ${voteDetail.associateDetail.cpf} already vote")
        }
    }
}